cpp/tests/copying/scatter_list_scalar_tests.cpp

```diff
@@ -0,0 +1,443 @@
+#include <cudf/null_mask.hpp>
+#include <cudf/scalar/scalar_factories.hpp>
+
+namespace cudf {
+namespace test {
+
+using mask_vector = std::vector<valid_type>;
+using size_column = fixed_width_column_wrapper<size_type>;
+
+class ScatterListScalarTests : public cudf::test::BaseFixture {
+};
+
+void test_single_scalar_scatter(column_view const& target,
+                                scalar const& slr,
+                                column_view const& scatter_map,
+                                column_view expect)
+{
+  std::vector<std::reference_wrapper<const scalar>> slrs{slr};
+  table_view targets{{target}};
+  auto result = scatter(slrs, scatter_map, targets, true);
+  CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(result->view().column(0), expect);
+}
+
+template <typename T>
+class ScatterListOfFixedWidthScalarTest : public ScatterListScalarTests {
+};
+
+TYPED_TEST_CASE(ScatterListOfFixedWidthScalarTest, FixedWidthTypesWithoutFixedPoint);
+
+// Test grid
+// Dim1 : {Fixed width, strings, lists, structs}
+// Dim2 : {Null scalar, Non-null empty scalar, Non-null non-empty scalar}
+// Dim3 : {Nullable target, non-nullable target row}
+
+TYPED_TEST(ScatterListOfFixedWidthScalarTest, Basic)
+{
+  using LCW = lists_column_wrapper<TypeParam, int32_t>;
+  using FCW = fixed_width_column_wrapper<TypeParam>;
+
+  auto slr = std::make_unique<list_scalar>(FCW({2, 2, 2}, {1, 0, 1}), true);
+  LCW col{LCW{1, 1, 1}, LCW{8, 8}, LCW{10, 10, 10, 10}, LCW{5}};
+  size_column scatter_map{3, 1, 0};
+
+  LCW expected{LCW({2, 2, 2}, mask_vector{1, 0, 1}.begin()),
+               LCW({2, 2, 2}, mask_vector{1, 0, 1}.begin()),
+               LCW{10, 10, 10, 10},
+               LCW({2, 2, 2}, mask_vector{1, 0, 1}.begin())};
+  test_single_scalar_scatter(col, *slr, scatter_map, expected);
+}
+
+TYPED_TEST(ScatterListOfFixedWidthScalarTest, EmptyValidScalar)
+{
+  using LCW = lists_column_wrapper<TypeParam, int32_t>;
+  using FCW = fixed_width_column_wrapper<TypeParam>;
+
+  auto slr = std::make_unique<list_scalar>(FCW{}, true);
+  LCW col{LCW{1, 1, 1},
+          LCW{8, 8},
+          LCW({10, 10, 10, 10}, mask_vector{1, 0, 1, 0}.begin()),
+          LCW{5},
+          LCW{42, 42}};
+  size_column scatter_map{1, 0};
+
+  LCW expected{
+    LCW{}, LCW{}, LCW({10, 10, 10, 10}, mask_vector{1, 0, 1, 0}.begin()), LCW{5}, LCW{42, 42}};
+  test_single_scalar_scatter(col, *slr, scatter_map, expected);
+}
+
+TYPED_TEST(ScatterListOfFixedWidthScalarTest, NullScalar)
+{
+  using LCW = lists_column_wrapper<TypeParam, int32_t>;
+  using FCW = fixed_width_column_wrapper<TypeParam>;
+
+  auto slr = std::make_unique<list_scalar>(FCW{}, false);
+  LCW col{LCW({1, 1, 1}, mask_vector{0, 0, 1}.begin()), LCW{8, 8}, LCW{10, 10, 10, 10}, LCW{5}};
+  size_column scatter_map{3, 1};
+
+  LCW expected({LCW({1, 1, 1}, mask_vector{0, 0, 1}.begin()), LCW{}, LCW{10, 10, 10, 10}, LCW{}},
+               mask_vector{1, 0, 1, 0}.begin());
+  test_single_scalar_scatter(col, *slr, scatter_map, expected);
```

Comment:
As commented above, the function returns the scatter result, then after this line we call
```
CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(result->view(), expect);
```